e1039-analysis/SimHits/macro/RecoE1039Sim.C

```diff
@@ -273,70 +293,71 @@ int Fun4Sim_Muons(const int nevent = 200,
+  reco->set_evt_reducer_opt("none");              // if not provided, event reducer will be using JobOptsSvc to intialize; to turn off, set it to "none", for normal tracking, set to something like "aoc"
+  reco->set_enable_eval(false);                   // set to true to generate evaluation file which includes final track candidates 
+  reco->set_eval_file_name("eval.root");          // evaluation filename
+  reco->set_enable_eval_dst(false);               // set to true to include final track candidates in the DST tree
+  //reco->add_eval_list(3);                         // include back partial tracks in eval tree for debuging
+  //reco->add_eval_list(2);                         // include station-3+/- in eval tree for debuging
+  //reco->add_eval_list(1);                         // include station-2 in eval tree for debugging
   se->registerSubsystem(reco);
 
-  // Vertex fitting for dimuon information
-  VertexFit *vertexing = new VertexFit();
-  se->registerSubsystem(vertexing);
-
-  // Truth node maker after tracking
-  TruthNodeMaker *truthMaker = new TruthNodeMaker();
-  // truthMaker->set_legacy_rec_container(legacy_rec_container);
-  truthMaker->Verbosity(verbosity);
-  if (do_aprime) {
-    // our signal input file seems to have different format than the offical one
-    // set m_do_evt_header to false so that TruthNodeMaker can run successfully
-    truthMaker->set_do_event_header(false);
+  // truth node maker after tracking
+  TruthNodeMaker* truthMaker = new TruthNodeMaker();
+  truthMaker->set_legacy_rec_container(legacy_rec_container);
+  if(do_aprime_muon or do_aprime_electron){
+    truthMaker->set_m_process_type(3); // set process type to 3 (A' -> di lepton) since we only have a 3 particle process instead of 0+1->2+3
   }
+  truthMaker->Verbosity(verbosity);
   se->registerSubsystem(truthMaker);
 
   // trigger emulator
-  // DPTriggerAnalyzer seems to need TruthNodeMaker
-  // to associate the trigger to SQEvent (?)
-  DPTriggerAnalyzer *dptrigger = new DPTriggerAnalyzer();
+  // needs TruthNodeMaker to associate the trigger to SQEvent 
+  DPTriggerAnalyzer* dptrigger = new DPTriggerAnalyzer();                                                                                                                                                 
   dptrigger->set_road_set_file_name("$E1039_RESOURCE/trigger/trigger_67.txt");
-  se->registerSubsystem(dptrigger);
+  dptrigger->Verbosity(verbosity);
+  se->registerSubsystem(dptrigger);  
 
-  // Analysis module
+  // event filter
+  EvtFilter *evt_filter = new EvtFilter();
+  evt_filter->Verbosity(verbosity);
+  evt_filter->set_trigger_req(1<<5);
+  //se->registerSubsystem(evt_filter);
+
+  // truth vertexing
+  SQTruthVertexing* truthVtx = new SQTruthVertexing();
+  truthVtx->set_legacy_rec_container(legacy_rec_container);
+  truthVtx->set_vtx_smearing(50.); // smear the truth z_vertex to mimic resolution effect, default is 0.
+  se->registerSubsystem(truthVtx);
```

Comment:
do we need this `SQTruthVertexing` for now?

Reply:
I think is useful for the case where we do not have real vertexing (for legacy_rec_container=false), but I can take out by default if that is better

Reply:
they seem to have some moduels that partially overlapped; https://github.com/E1039-Collaboration/e1039-core/blob/master/simulation/g4dst/TruthNodeMaker.cc#L347-L351 this TruthNodeMaker would produce `SQTruthDimuonVector` as well?

Reply:
Interesting, yeah so it seems that SQTruthVertexing will not produce SQTruthDimuonVector but actually use it to associate with the tracks assuming some smearing of the vertex.
https://github.com/liuk/e1039-core/blob/vtxreco/packages/reco/ktracker/SQTruthVertexing.h#L22-L36

This means that it will produce a "rec dimuon" collection for the case where legacy_rec_container=false that can't run the vertexing fit yet. 

But since it seems that we are sticking with legacy_rec_container=true I can comment it out for now. Does that sound good?